pkg/kubelet/container/helpers.go

```diff
@@ -104,3 +108,47 @@ func ExpandContainerCommandAndArgs(container *api.Container, envs []EnvVar) (com
 
 	return command, args
 }
+
+// Create an event recorder to record object's event except implicitly required container's, like infra container.
+func FilterEventRecorder(recorder record.EventRecorder) record.EventRecorder {
+	return &innerEventRecorder{
+		recorder: recorder,
+	}
+}
+
+type innerEventRecorder struct {
+	recorder record.EventRecorder
+}
+
+func (irecorder *innerEventRecorder) Event(object runtime.Object, reason, message string) {
+	if object == nil {
+		return
+	}
+	if ref, ok := object.(*api.ObjectReference); ok {
+		if !strings.HasPrefix(ref.FieldPath, ImplicitContainerPrefix) {
```

Comment:
The code block from line 124 to 128 repeats in every method. Suggest creating a method to reduce duplication.